### DISCUSSION TITLE
Feature request: time picker 'today since 7:00 AM'

### DISCUSSION THREAD

**Discussion by wz2b** [grafana/grafana#37760] [Platform]
The time picker selection "Today So Far" is fantastic - it is by far what I use most often.

For many applications, there is often a useless gap the beginning of the day.  This could be because nobody is working yet in my time zone, so I don't really want to see midnight until now, I want to see  'start of work day' until now.  A second example of a case where this is really useful is solar monitoring - the solar panels produce nothing between midnight and sunrise.

You can't really work around this using "now".  A member of the Slack community and I were talking and we were spit-balling ideas, but don't know if any of these are practical:

 * Be able to specify from/to in the time picker as just a time, with no date, and it infers you meant today
 * Have some way to do math on 'now' to extract just the date part and specify the 'from' time as 'datepart(now)+6h'

Again, we weren't really sure what was a practical way to do this, but the 'gap since midnight' was something I raised and other people stepped up and said "Yes, I would like that to" so I figured it was worth a feature request.

**Comment by zswanson** [comment]
Similar use-case for my company's product as we are very east coast business hours for all our traffic. Most of our overnight data is of no interest.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
